ocaml/xapi/xapi.ml

```diff
@@ -924,6 +922,9 @@ let server_init() =
           "Synchronising network sriovs on slave with master", [Startup.OnlySlave; Startup.NoExnRaising], Sync_networking.copy_network_sriovs_from_master ~__context;
           "Synchronising VLANs on slave with master", [Startup.OnlySlave; Startup.NoExnRaising], Sync_networking.copy_vlans_from_master ~__context;
           "Synchronising tunnels on slave with master", [Startup.OnlySlave; Startup.NoExnRaising], Sync_networking.copy_tunnels_from_master ~__context;
+          (* CA-290237: Create cluster objects after network objects so PIFs are available *)
+          "Create any necessary cluster_host objects", [ Startup.NoExnRaising ],
```

Comment:
@minishrink we'll need to move this after we got an IP. Creating a cluster host object checks for the presence of an IP, and you get a NO_COMPATIBLE_CLUSTER_HOST error, caused by the create failing due to PIF_HAS_NO_NETWORK_CONFIGURATION

Reply:
Creating the object not only creates the object in the Db but also tries to enable it, and creates the object in the db only if enabling succeeded.

Reply:
Please add CA-290473 in the commit message when this is fixed.

Reply:
Actually it might be better to have a function here just create the database object, and have the actual join/enable done elsewhere (in resync_host).
For that to work:
* add a new field to cluster_host called joined, default to true for backwards compat
* refactor `Cluster_host.create` into a create_db and create (where create calls create_db and then resync_host)
* `resync_host` will look at the joined field: if true then it calls daemon.enable+join, and sets joined to true if successful, otherwise it calls enable if enabled
* in the startup sequence here only call create_db